show-labels-on-layer/src/main/java/com/esri/arcgismaps/sample/showlabelsonlayer/MainActivity.kt

```diff
@@ -0,0 +1,146 @@
+import com.arcgismaps.mapping.labeling.LabelDefinition
+import com.arcgismaps.mapping.layers.FeatureLayer
+import com.arcgismaps.mapping.symbology.TextSymbol
+import com.esri.arcgismaps.sample.showlabelsonlayer.databinding.ActivityMainBinding
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.launch
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    // set up data binding for the activity
+    private val activityMainBinding: ActivityMainBinding by lazy {
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    private val mapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // create a map with a light gray map style
+        val map = ArcGISMap(BasemapStyle.ArcGISLightGray)
+        // create a service feature table from an online feature service of
+        // US Congressional Districts
+        val serviceFeatureTable =
+            ServiceFeatureTable(getString(R.string.congressional_districts_url))
+        // create the feature layer from the service feature table
+        val featureLayer = FeatureLayer(serviceFeatureTable)
+        // add this feature layer to the map
+        map.operationalLayers.add(featureLayer)
+        // add the map to the mapview
+        mapView.map = map
+
+        lifecycleScope.launch {
+            // if the map load failed show an error and return
+            map.load().onFailure {
+                return@onFailure showError("Error loading map:${it.message}")
+            }
+            // if the feature layer load failed show an error and return
+            featureLayer.load().onFailure {
+                return@onFailure showError("Error loading feature layer:${it.message}")
+            }
+            // zoom to the layer when it's done loading
+            featureLayer.fullExtent?.let { extent ->
+                mapView.setViewpoint(Viewpoint(extent))
+            }
+        }
+
+        // create label definitions for the different party type attribute in the feature service
+        val republicanLabelDefinition = createLabelDefinition("Republican", Color.red)
+        val democraticLabelDefinition = createLabelDefinition("Democrat", Color.blue)
+
+        featureLayer.apply {
+            // add the label definitions to the feature layer
+            labelDefinitions.addAll(
+                listOf(
+                    republicanLabelDefinition,
+                    democraticLabelDefinition
+                )
+            )
+            // enable the labels
+            featureLayer.labelsEnabled = true
```

Comment:
The `featureLayer` qualification shouldn't be needed since this is in the apply block